ipa-core/src/ff/boolean_array.rs

```diff
@@ -254,9 +256,15 @@ macro_rules! boolean_array_impl {
     type Store = BitArr!(for $bits, in u8, Lsb0);
 
             /// A Boolean array with $bits bits.
-            #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+            #[derive(Clone, Copy, PartialEq, Eq)]
             pub struct $name(pub(super) Store);
 
+            impl Debug for $name {
+                fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+                    write!(f, "{}({})", stringify!($name), self.0)
```

Comment:
does the compiler know how to do this:

```suggestion
                    write!(f, concat!(stringify!($name), "({})"), self.0)
```

Reply:
Separately, it strikes me that the brackets might be enough:

```suggestion
                    write!(f, "{}{}", stringify!($name), self.0)
```

Reply:
Which leads to:

```suggestion
                    f.write_str(stringify!($name))?;
                    self.0.fmt(f)
```